### PR TITLE
[FLINK-18797][docs][examples] Update deprecated forms of keyBy in docs and examples

### PR DESCRIPTION
**Pull request by wulei0302**
## What is the purpose of the change

The `keyBy` usage of the examples and documents is no longer recommended, so update to the latest usage. This PR only updates docs and examples.

## Brief change log

[a2a8717](https://github.com/apache/flink/commit/a2a8717c3738c883d0dfadf3f4b59289302bca21) : In the examples module, if it is java class, update '.keyBy("word")' to '.keyBy(value -> value.word)', update '.keyBy(0)' to '.keyBy(value -> value.f0)'; if it is scala class, update '.keyBy("word")' to '.keyBy(_.word)', update '.keyBy(0)' to '.keyBy(_._1)';
[30d841e](https://github.com/apache/flink/commit/30d841ee0c5df64c98e1e572937427ccbac592b0) : update docs module like examples module
[3b2c0da](https://github.com/apache/flink/commit/3b2c0da11cb0ec78cf24b14c0b70a88df7fff745) : adjust `etl.md` code comment

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? no

### CODE REVIEW

**Comment by wulei0302** [comment]
> ## CI report:
> * [3b2c0da](https://github.com/apache/flink/commit/3b2c0da11cb0ec78cf24b14c0b70a88df7fff745) Azure: [FAILURE](https://dev.azure.com/apache-flink/98463496-1af2-4620-8eab-a2ecc1a2e6fe/_build/results?buildId=5475)
> 
> Bot commands
Sorry. This error is because of Idea's Optimize imports, I have fixed it

**Comment by wulei0302** [comment]
@wuchong hello, jark, Can you please review the code for me

File: docs/dev/types_serialization.md [hunk 1]
```diff
@@ -86,11 +86,11 @@ val input = env.fromElements(
     WordCount("hello", 1),
     WordCount("world", 2)) // Case Class Data Set
 
-input.keyBy("word")// key by field expression "word"
+input.keyBy(_.word) // .keyBy("word") is @Deprecated
```
Comment (dawidwys) [inline docs/dev/types_serialization.md:89]: In my opinion, there is no point in mentioning something that is deprecated. Let's just drop the comment.

File: docs/dev/types_serialization.md [hunk 2]
```diff
@@ -137,7 +137,7 @@ DataStream<WordWithCount> wordCounts = env.fromElements(
     new WordWithCount("hello", 1),
     new WordWithCount("world", 2));
 
-wordCounts.keyBy("word"); // key by field expression "word"
+wordCounts.keyBy(value -> value.word); // .keyBy("word") is @Deprecated
```
Comment (dawidwys) [inline docs/dev/types_serialization.md:140]: In my opinion, there is no point in mentioning something that is deprecated. Let's just drop the comment.

File: docs/dev/types_serialization.md [hunk 3]
```diff
@@ -86,11 +86,11 @@ val input = env.fromElements(
     WordCount("hello", 1),
     WordCount("world", 2)) // Case Class Data Set
 
-input.keyBy("word")// key by field expression "word"
+input.keyBy(_.word) // .keyBy("word") is @Deprecated
 
 val input2 = env.fromElements(("hello", 1), ("world", 2)) // Tuple2 Data Set
 
-input2.keyBy(0, 1) // key by field positions 0 and 1
+input2.keyBy(value => (value._1, value._2)) // key by field positions 0 and 1
```
Comment (dawidwys) [inline docs/dev/types_serialization.md:93]: We don't key by positions here. Let's remove the comment.

File: docs/dev/types_serialization.md [hunk 4]
```diff
@@ -153,7 +153,7 @@ val input = env.fromElements(
     new WordWithCount("hello", 1),
     new WordWithCount("world", 2)) // Case Class Data Set
 
-input.keyBy("word")// key by field expression "word"
+input.keyBy(_.word) // .keyBy("word") is @Deprecated
```
Comment (dawidwys) [inline docs/dev/types_serialization.md:156]: In my opinion, there is no point in mentioning something that is deprecated. Let's just drop the comment.

File: docs/dev/types_serialization.md [hunk 5]
```diff
@@ -238,7 +238,7 @@ Flink tries to infer a lot of information about the data types that are exchange
 Think about it like a database that infers the schema of tables. In most cases, Flink infers all necessary information seamlessly
 by itself. Having the type information allows Flink to do some cool things:
 
-* Using POJOs types and grouping / joining / aggregating them by referring to field names (like `dataSet.keyBy("username")`).
+* Using POJOs types and grouping / joining / aggregating them by referring to field names (like `dataSet.keyBy(value -> value.username)`).
```
Comment (dawidwys) [inline docs/dev/types_serialization.md:241]: We are no longer referring to field names here. Honestly I am not sure if the point makes sense with the `KeySelector` only. The way I read this point is tells you can use field names, which we discourage nowadays.

I'd rather remove the point whatsoever.

File: docs/learn-flink/etl.md [hunk 6]
```diff
@@ -149,6 +149,7 @@ this would mean doing some sort of GROUP BY with the `startCell`, while in Flink
 {% highlight java %}
 rides
     .flatMap(new NYCEnrichment())
+    // .keyBy("fieldname") is @Deprecated, could use .keyBy(value -> value.startCell)
```
Comment (dawidwys) [inline docs/learn-flink/etl.md:152]: Why don't we just use the correct version instead? Again I can't see a reason why we should even mention deprecated methods.

**Review by dawidwys — CHANGES_REQUESTED** [review]
Thank you @wulei0302 for the PR! I added some minor suggestions. Could you also squash the last two commits into respective previous ones?

File: docs/dev/stream/operators/index.md [hunk 7]
```diff
@@ -104,8 +104,8 @@ dataStream.filter(new FilterFunction<Integer>() {
             <p>
             This transformation returns a <em>KeyedStream</em>, which is, among other things, required to use <a href="{{ site.baseurl }}/dev/stream/state/state.html#keyed-state">keyed state</a>. </p>
 {% highlight java %}
-dataStream.keyBy("someKey") // Key by field "someKey"
-dataStream.keyBy(0) // Key by the first element of a Tuple
+dataStream.keyBy(value -> value.someKey) // Key by field "someKey", or maybe use getSomeKey
```
[the remark this answers is not in the document]
Comment (wulei0302) [reply docs/dev/stream/operators/index.md]: @dawidwys  Thank you very much for your review. I am very happy for your good suggestions.  Please help me review it again.  thank you.

**Comment by dawidwys** [comment]
Hey @wulei0302 Could you please fix the PR's commit history. Right now there is a somewhat weird merge commit. Could you please fix it so that it is linear. Thx

**Comment by wulei0302** [comment]
Hey @dawidwys  Sorry to take your time, because this is my first time committing in the community, so thank you very much for your guidance. I have corrected it, and please help me see it again

File: docs/dev/connectors/cassandra.md [hunk 8]
```diff
@@ -231,7 +231,7 @@ DataStream<WordCount> result = text
                 }
             }
         })
-        .keyBy("word")
+        .keyBy(WordCount :: getWord)
```
Comment (dawidwys) [inline docs/dev/connectors/cassandra.md:234]: I will remove the whitespaces while merging though.
```suggestion
        .keyBy(WordCount::getWord)
```

File: docs/dev/connectors/cassandra.zh.md [hunk 9]
```diff
@@ -231,7 +231,7 @@ DataStream<WordCount> result = text
                 }
             }
         })
-        .keyBy("word")
+        .keyBy(WordCount :: getWord)
```
Comment (dawidwys) [inline docs/dev/connectors/cassandra.zh.md:234]: I will remove the whitespaces while merging though.
```suggestion
        .keyBy(WordCount::getWord)
```

**Review by dawidwys — APPROVED** [review]
No worries ;) Thank you for the update and the PR.

I had one last minor comment that I will address while merging. Thanks again.

**Pull request closed without merging**